gneiss/plot/_dendrogram.py

```diff
@@ -0,0 +1,235 @@
+
+        # reorder so that x and y are first
+        cols = list(result)
+        cols.insert(0, cols.pop(cols.index('y')))
+        cols.insert(0, cols.pop(cols.index('x')))
+        result = result.ix[:, cols]
+        return result
+
+    @abc.abstractmethod
+    def rescale(self, width, height):
+        pass
+
+
+class UnrootedDendrogram(Dendrogram):
+    aspect_distorts_lengths = True
+
+    def __init__(self, **kwargs):
+        """ Constructs a UnrootedDendrogram object for visualization.
+
+        Parameters
+        ----------
+        use_lengths: bool
+            Specifies if the branch lengths should be included in the
+            resulting visualization (default True).
+        """
+        super().__init__(**kwargs)
+
+    @classmethod
+    def from_tree(cls, tree):
+        """ Creates an UnrootedDendrogram object from a skbio tree.
+
+        Parameters
+        ----------
+        tree : skbio.TreeNode
+            Input skbio tree
+
+        Returns
+        -------
+        UnrootedDendrogram
+        """
+        for n in tree.postorder(include_self=True):
+            n.__class__ = UnrootedDendrogram
+        tree._cache_ntips()
+        return tree
+
+    def rescale(self, width, height):
+        """ Find best scaling factor for fitting the tree in the canvas
+
+        Parameters
+        ----------
+        width : float
+            width of the canvas
+        height : float
+            height of the canvas
+
+        Returns
+        -------
+        best_scaling : float
+            largest scaling factor in which the tree can fit in the canvas.
+
+        Notes
+        -----
+        This function has a little bit of recursion.  This will
+        need to be refactored to remove the recursion.
+        """
+
+        angle = 2*numpy.pi / self._n_tips
+        # this loop is a horrible brute force hack
+        # there are better (but complex) ways to find
+        # the best rotation of the tree to fit the display.
+        best_scale = 0
+        for i in range(60):
+            direction = i / 60.0 * numpy.pi
+            points = self.update_coordinates(1.0, 0, 0, direction, angle)
+            xs = [x for (x, y) in points]
+            ys = [y for (x, y) in points]
+
+            # double check that the tree fits within the margins
+            scale = min(float(width) / (max(xs) - min(xs)),
+                        float(height) / (max(ys) - min(ys)))
+            scale *= 0.95  # extra margin for labels
+            if scale > best_scale:
+                best_scale = scale
+                mid_x = width / 2 - ((max(xs) + min(xs)) / 2) * scale
+                mid_y = height / 2 - ((max(ys) + min(ys)) / 2) * scale
+                best_args = (scale, mid_x, mid_y, direction, angle)
+
+        self.update_coordinates(*best_args)
+        return best_scale
+
+    def update_coordinates(self, s, x1, y1, a, da):
+        """ Update x, y coordinates of tree nodes in canvas.
```

Comment:
Two scenarios

1. This is called within the `rescale` function, where the best orientation / scaling is chosen to fit the tree in the proper dimensions
2. Dynamic rendering of wedges, so that we can represent some of the subtrees as wedges